src/types/jobs/JobSchedule.type.ts

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export enum FixedPeriod {
+  Yearly = "@yearly",
+  Monthly = "@monthly",
+  Weekly = "@weekly",
+  Daily = "@daily",
+  Hourly = "@hourly",
+}
+
+type EveryPeriod = `@every ${string}`;
+
+// note: This can get crazy, more than TS can really handle.
```

Comment:
You might consider taking a look at [how I built this out](https://github.com/dapr/dotnet-sdk/blob/master/src/Dapr.Jobs/Models/DaprJobSchedule.cs) in the .NET SDK to accommodate each variation.

Reply:
Most importantly - there are different sets of arguments that need to be provided based on this schedule and the other arguments used.